pxtlib/tutorial.ts

```diff
@@ -305,6 +306,27 @@ ${code}
         return { header, hint, requiredBlocks };
     }
 
+    function categorizingValidationRules(listOfRules: pxt.Map<boolean>) {
+        const ruleNames: string[] = Object.keys(listOfRules);
```

Comment:
nit: for the most part it's fine / preferable to omit explicit typings like these, as you get the same from type inference / it's often just extra noise.